examples/samples/deferral.ts

```diff
@@ -0,0 +1,123 @@
+import {
+  OnMessage,
+  OnError,
+  MessagingError,
+  delay,
+  ServiceBusMessage,
+  ReceiveMode,
+  generateUuid,
+  Namespace,
+  SendableMessageInfo
+} from "../../lib";
+import * as dotenv from "dotenv";
+dotenv.config();
+
+const str = process.env.SERVICEBUS_CONNECTION_STRING || "";
+const path = process.env.QUEUE_NAME || "";
+console.log("str: ", str);
+console.log("path: ", path);
+
+async function main(): Promise<void> {
+  await sendMessage();
+  await receiveMessage();
+}
+
+async function sendMessage(): Promise<void> {
+  const nsSend = Namespace.createFromConnectionString(str);
+  const sendClient = nsSend.createQueueClient(path);
+  var data = [
+    { step: 1, title: "Shop" },
+    { step: 2, title: "Unpack" },
+    { step: 3, title: "Prepare" },
+    { step: 4, title: "Cook" },
+    { step: 5, title: "Eat" }
+  ];
+  try {
+    var promises = new Set<Promise<void>>();
+    for (let index = 0; index < data.length; index++) {
+      const message: SendableMessageInfo = {
+        body: JSON.stringify(data[index]),
+        label: "RecipeStep",
+        contentType: "application/json",
+        timeToLive: 2 * 60 * 1000, // 2 minutes
+        messageId: generateUuid()
```

Comment:
I agree with you but I believe these are the basic properties people should know about. The sample may be about queue deferral but we should overselves strive to follow best practice. 
IMO, messageId should be set for uniqueness, and timeToLive should be set otherwise the message would fall back to default value set for the queue which could be confusing to the user.
@AlexGhiondea  @ramya-rao-a , what do you think ?

Reply:
I'd still prefer to mention these two properties for two reasons:
- I feel that timeToLive is an important concept as well and I'd like user to think about it before creating a message. If a user doesn't specify the timeToLive, the message expiration time would fall back to the default value specified by queue and I think it could lead to other issues in the queue. That's why I thought that lets put it here as a best practice.
- I feel that we should also have a consistent format across our samples. If we've specified these properties before, I think its a good practice to follow the convention otherwise it'd look awkward and could potentially confuse customer to think that message deferral only works when we don't specify these two (messageId, timeToLive) properties.

I agree with you that the we should provide a simple examples for our users, but I personally don't think adding these two properties is making it complicated or so. If you feel, it is confusing for the user we can also add comment like "timeToLive is not a mandatory field". 

Overall, I don't have any problem with going either way, and I can make the change if you feel strongly about this change.
@AlexGhiondea what are your thoughts about this?

Reply:
Yes, consistency is important. We can go ahead with what we have now.
When we do a review of all our samples in the end, we can take a call on this matter